src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1767,11 +1865,15 @@ bool entitiesQuery
       // FIXME P4: Once issue #1705 is implemented, this check can be removed.
       if (qStringFilters(sco->value, filters) != true)
       {
-        if (badInputP)
+        if (badInputP)  // Bad Input reported by higher level
         {
           *badInputP = true;
           *err       = "invalid query expression";
         }
+        else
```

Comment:
I understand that in that case, the caller would manage to set a not-NULL badInputP and manage the alarm itself.

If this else is not needed at the end, I'd suggest to remove it, for not adding extra entropy not related with the fix that the PR is addressing. At the end, any potential imprevement/fix in the bad input stuff will be covered by the issue https://github.com/telefonicaid/fiware-orion/issues/1763